crates/metadata/src/specs.rs

```diff
@@ -994,6 +1028,46 @@ where
     }
 }
 
+/// This is wrapper for [`TypeSpec`] which acts as a factory.
+/// The whole purpose of the factory is to replace the type of
+/// `Ok` variant of `Result` with `()` because constructors do not return
+/// any data upon successful instantiation.
+///
+/// # Important Note
+/// Only use this factory with constructors!
+pub trait TransformType {
```

Comment:
I find the name of the Trait (and the trait method) confusing. Proposal in #1473